tools/collect-sources.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+# collect-sources.sh
+# collect sources for alpine, go and kernel packages used in eve
+# arg1=tarfile with rootfs.tar
+# arg2=path to eve root
+# arg3=path to tar.gz file for output. If blank, will send to stdout
+
+set -e
+
+rootfs="$1"
+eve="$2"
+outfile="$3"
+
+[ -z "$outfile" ] && outfile=-
+
+tmproot=$(mktemp -d)
+tmpout=$(mktemp -d)
+manifest=${tmpout}/collected_sources_manifest.csv
+
+# this is a bit of a hack, but we need to extract the rootfs tar to a directory, and it fails if
+# we try to extract character devices, block devices or pipes, so we just exclude the dir.
+tar -xf "$rootfs" -C "$tmproot" --exclude "dev/*"
+
+{
+"${eve}/tools/get-alpine-pkg-source.sh" -s "${tmpout}" -e "${tmproot}" -p alpine
+"${eve}/tools/get-kernel-source.sh" -s "${tmpout}" -p kernel
```

Comment:
Can't you call get-kernel-source.sh -s "${tmpout}/kernel" and not need the additional argument?

Reply:
See above.